clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -97,7 +97,25 @@ public static RecordingLevel forName(String name) {
         public boolean shouldRecord(final int configId) {
             return configId == DEBUG.id || configId == this.id;
         }
+    }
 
+    public enum QuotaEnforcementType {
+        /**
+         * The quota is not enforced.
+         */
+        NONE,
+
+        /**
+         * The quota is enforced before updating the sensor. An update
+         * is rejected if the quota is already exhausted.
```

Comment:
nit: somehow the first sentence is confusing because it seem to imply that the sensor gets updated in this case as well. Do you think maybe this sounds a bit clearer? "The check for quota violation is done prior to recording the value. If the quota is already exhausted, the value is not recorded into the sensor." Or perhaps some other  Up to you.

Reply:
That is indeed much better.